src/app.js

```diff
@@ -181,10 +224,60 @@ gpii.app.exit = function (that) {
 };
 
 /**
-  * Handles when a user token is keyed out through other means besides the task tray key out feature.
-  * @param that {Component} An instance of gpii.app
-  * @param keyedOutUserToken {String} The token that was keyed out.
-  */
+ * Listens to any changes in the lifecycle managers logonChange model and calls the
+ * appropriate functions in gpii-app for notifying of the current user state
+ */
+gpii.app.logonChangeListener = function (that, model) {
+    model.inProgress ? gpii.app.displayWaitDialog(that) : gpii.app.dismissWaitDialog(that);
+};
+
+/**
+ * Shows the dialog on users screen with the message passed as parameter.
+ * Records the time it was shown in `dialogStartTime` which we need when
+ * dismissing it (checking whether it's been displayed for the minimum amount of time)
+ *
+ * @param that {Object} the app module
+ */
+gpii.app.displayWaitDialog = function (that) {
+    that.dialog.show();
+    // Hack to ensure it stays on top, even as the GPII autoconfiguration starts applications, etc., that might
```

Comment:
@kaspermarkus I'm not sure this is doing what you intended. The on screen keyboard is placed on top of the wait dialog. Perhaps this isn't something we can control.